src/opaque-types-type-alias-impl-trait-inference.md

```diff
@@ -0,0 +1,193 @@
+# Inference of opaque types (type alias `impl Trait`)
+
+This page describes how the compiler infers the hidden type for an opaque type. This kind of type inference is particularly complex because, unlike other kinds of type inference, it works across functions and function bodies.
```

Comment:
```suggestion
This page describes how the compiler infers the [hidden type] for an [opaque type]. This kind of type inference is particularly complex because, unlike other kinds of type inference, it works across functions and function bodies.

[hidden type]: https://rustc-dev-guide.rust-lang.org/borrow_check/region_inference/member_constraints.html
[opaque type]: https://rustc-dev-guide.rust-lang.org/opaque-types-type-alias-impl-trait.html
```